jetty-core/jetty-http2/jetty-http2-common/src/main/java/org/eclipse/jetty/http2/HTTP2Channel.java

```diff
@@ -34,6 +35,8 @@ public interface HTTP2Channel
      */
     public interface Client
     {
+        public Runnable onHeaders(Stream stream, HeadersFrame frame, Callback callback);
```

Comment:
There should be javadoc explaining how the returned runnables are to be executed as sometimes it's via the execution strategy, sometimes it's via `executeImmediately`.

It's paramount to know which is executed how to make sure the implementation returns invocable runnables where needed.